### ISSUE TITLE
Vendor "node-types.json" and "grammar.json"

### ISSUE THREAD

**Issue by ahumenberger** [smacker/go-tree-sitter#95]
We are creating an "AST generator" generator, which generates AST node types from the tree-sitter `node-types.json` file (as in https://github.com/tree-sitter/tree-sitter-java/tree/master/src). For this it would be great to have the `node-types.json` file next to the actual `parser.c` file to make sure to always have the correct version in place. And at some point we will also need the `grammar.json` file.

So long story short, I'd like to adapt the vendoring script such that also those JSON files are considered. Would such a change be accepted?

**Comment by smacker** [comment]
I think it would be better if you vendor `node-types.json` to your own project that needs those files.

Otherwise, we will make users of `go-tree-sitter` download 2mb+ (75kb * 27 grammars + the files change on grammars update) while those files are not required for the library to work.

**Comment by ahumenberger** [comment]
That's okay, I guess. This would require some changes in the vendoring nonetheless. As of now, if a language is vendored via a branch and not with a specific tag/version, then it is not known which revision is currently in use, or is it? But that's crucial for us to vendor the correct `node-types.json`.

So I propose to add a `vendor-languages.json` (or similar) containing entries like 

```json
	{
		"language": "kotlin",
		"url": "https://github.com/fwcd/tree-sitter-kotlin",
		"reference": "main",
		"revision": "e1cd1a0a520837806993afa1f6fd4e430010f6d7",
	},
	{
		"language": "java",
		"url": "https://github.com/tree-sitter/tree-sitter-java",
		"reference": "v0.20.1",
		"revision": "e1cd1a0a520837806993afa1f6fd4e430010f6d7",
	}
```

The vendoring script then updates this JSON file.

**Comment by smacker** [comment]
@ahumenberger sounds good
